codec-http/src/main/java/io/netty/handler/codec/http/HttpServerUpgradeHandler.java

```diff
@@ -284,7 +284,21 @@ private boolean upgrade(final ChannelHandlerContext ctx, final FullHttpRequest r
         }
 
         // Make sure the CONNECTION header is present.
-        CharSequence connectionHeader = request.headers().get(HttpHeaderNames.CONNECTION);
+        List<String> connectionHeaderValues = request.headers().getAll(HttpHeaderNames.CONNECTION);
+
+        if (connectionHeaderValues == null) {
+            return false;
+        }
+
+        CharSequence connectionHeader = null;
+
+        for (CharSequence connectionHeaderValue : connectionHeaderValues) {
+            if (containsIgnoreCase(connectionHeaderValue, HttpHeaderNames.UPGRADE)) {
+                connectionHeader = connectionHeaderValue;
```

Comment:
This requires that Upgrade exists in the same Connection header value as requiredHeaders. But each value could be put in its own Connection header. It would be better to join all the values together with a `,` and then splitHeader(). @normanmaurer, is there not a utility to do that?

Reply:
@ejona86 thats a good point... I think there is no such utility yet (or at least I could not remember and also could not find any). /cc @arukshani can you do the adjustment and also add a unit test for this ?

Reply:
@ejona86 and @normanmaurer I have done the necessary changes and added a unit test as well for that. Please review.